Libraries/Lists/FlatList.js

```diff
@@ -518,8 +520,8 @@ class FlatList<ItemT> extends React.PureComponent<Props<ItemT>, void> {
     }
   };
 
-  _getItemCount = (data: ?Array<ItemT>): number => {
-    if (Array.isArray(data)) {
+  _getItemCount = (data: ?$ArrayLike<ItemT>): number => {
+    if (data != null && typeof Object(data).length === 'number') {
```

Comment:
If it's ArrayLike obj and not null, it must have a length prop defined. ?
And we don't want those with only _size_ defined or do we ?
any specific reason for new Object instance _overhead_?
```suggestion
    if (data != null && typeof data.length === 'number') {
```

Reply:
Per MDN this should return the existing object if it is one, but the check would cause an exception if someone passed say, a number, and the point here has been to guard against invalid data gracefully for legacy reasons.